api/v1/db/json/notes-dao.js

```diff
@@ -0,0 +1,93 @@
+const appRoot = require('app-root-path');
+const _ = require('lodash');
+const fs = require('fs');
+const config = require('config');
+
+const { serializeNotes } = require('../../serializers/notes-serializer');
+
+/**
+ * @summary Return a list of notes filtered/sorted by query parameters
+ * @function
+ * @param {Object} query Query parameters
+ * @returns {Promise} Promise object represents a list of notes
+ */
+const getNotes = query => new Promise((resolve, reject) => {
+  try {
+    const { dbDirectoryPath } = config.api;
+    if (!fs.existsSync(`${appRoot}${dbDirectoryPath}`)) {
+      reject(new Error(`DB directory path: '${dbDirectoryPath}' is invalid`));
+    }
+
+    const {
+      studentID, q, sortKey, sources, contextTypes,
+    } = query;
+    const dirPath = `${dbDirectoryPath}/${studentID}`;
+    let files;
+    let rawNotes = [];
+    try {
+      files = fs.readdirSync(`${appRoot}${dirPath}`);
```

Comment:
Same comment I left in the config repo:

> I guess it's a bit confusing. Since you're using appRoot to concatenate the path so that this folder will be under the project home directory. However, it's not necessary to always storing data at the same project directory, so I think we should use the absolute path here.